cmd/lakectl/cmd/fs.go

```diff
@@ -115,28 +115,29 @@ var fsCatCmd = &cobra.Command{
 	Short: "Dump content of object to stdout",
 	Args:  cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		client := getClient()
 		pathURI := MustParsePathURI("path", args[0])
 		direct, _ := cmd.Flags().GetBool("direct")
-		var contents []byte
+		var err error
+		var body io.ReadCloser
+		client := getClient()
 		if direct {
-			_, body, err := helpers.ClientDownload(cmd.Context(), client, pathURI.Repository, pathURI.Ref, *pathURI.Path)
-			if err != nil {
-				DieErr(err)
-			}
-			defer body.Close()
-			contents, err = io.ReadAll(body)
-			if err != nil {
-				DieErr(err)
-			}
+			_, body, err = helpers.ClientDownload(cmd.Context(), client, pathURI.Repository, pathURI.Ref, *pathURI.Path)
 		} else {
-			resp, err := client.GetObjectWithResponse(cmd.Context(), pathURI.Repository, pathURI.Ref, &api.GetObjectParams{
+			var resp *http.Response
+			resp, err = client.GetObject(cmd.Context(), pathURI.Repository, pathURI.Ref, &api.GetObjectParams{
 				Path: *pathURI.Path,
 			})
 			DieOnResponseError(resp, err)
-			contents = resp.Body
+			body = resp.Body
+		}
+		if err != nil {
+			DieErr(err)
+		}
```

Comment:
You can move this one to the first 'if' block as this is no longer common check - the second part checks for error using 'DieOnResponseError'.

Reply:
Fixed